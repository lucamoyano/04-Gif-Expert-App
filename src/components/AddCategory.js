import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = ({ target: { value } }) => {
        setInputValue( value );
    }

    const handleSubmit = (e) => {
        e.preventDefault(); //Prevenir que se refresque la pagina
        
        if( inputValue.trim().length > 2 ) {
            setCategories( c => [inputValue, ...c]);
            setInputValue('');
        }
    }

    return (
        <form onSubmit={ handleSubmit }>
            <input 
                type="text" 
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}