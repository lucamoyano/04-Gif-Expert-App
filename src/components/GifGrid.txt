import React from 'react'
import {GifGridItem} from './GifGridItem';
import { getGifs } from '../helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ( {category} ) => {

    //const [images, setImages] = useState([]);

    // useEffect(() => {
    //     getGifs( category )
    //         //.then( imgs => setImages( imgs ) ); 
    //         //Como es una funcion que envia su primer argumento como primer argumento a la funcion
    //         //Se puede utilizar de la siguiente manera:
    //         .then( setImages );
    // }, [category]); 
    // //Le pasamod un arreglo vacio como segundo argumento
    // //Para decirle que se ejecute una sola vez

    const { data, loading } = useFetchGifs();

    return (
        <>
        <h3>{category}</h3>
        <div className="card-grid">
            {/* {
                images.map( ({id, title}) => (
                        <li key={id}> {title} </li>
                    )
                )
            } */}

            {/* {
                images.map( img => (
                    // <GifGridItem img={ img } />
                    <GifGridItem 
                        key={img.id}
                        { ...img }  //Le paso las props mediante desetructuracion
                    />
                    )
                )
            } */}
        </div>
        </>
    )
}
