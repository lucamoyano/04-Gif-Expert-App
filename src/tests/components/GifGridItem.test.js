import { GifGridItem } from '../../components/GifGridItem';
import { shallow } from 'enzyme';

describe('Pruebas en componente GifGridItem', () => {

    const title = 'Un titulo';
    const url = 'https://url.com';
    let wrapper = shallow( <GifGridItem title={title} url={url} />  );

    test('debe mostrar el componente correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    });
    
    test('debe tener un pÃ¡rrafo con el title', () => {
        const p = wrapper.find('p').text().trim();
        expect(p).toBe(title);
    });

    test('debe tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img');
        
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
    });

    test('debe tener animate__fadeIn', () => {
        const div = wrapper.find('div');
        let b = div.prop('className');

        expect (b.includes('animate__fadeIn')).toBe(true);
    });

});