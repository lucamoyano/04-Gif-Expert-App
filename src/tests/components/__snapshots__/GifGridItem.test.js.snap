// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente GifGridItem debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://url.com"
  />
  <p>
    Un titulo
  </p>
</div>
`;
